omnibus/config/software/datadog-agent.rb

```diff
@@ -16,7 +16,9 @@
 license "Apache-2.0"
 license_file "../LICENSE"
 
-source path: '..'
+source path: '..',
+       # Ignore folders holding big unused artifacts to speed up the copy
+       options: { exclude: ['bin', 'omnibus/pkg', 'Dockerfiles'] }
```

Comment:
Not sure I follow what we&omnibus are doing here: why do these artifacts end up in the parent directory in the first place?

Reply:
Instead of building the agent in place in the given source path, omnibus seems to want to copy everything to a new folder. This is pretty expensive, especially if the sources are in a virtualbox shared folder.

The idea is to speed it up by omitting:
  - `bin/`: that may contain locally-build agent/dogstatsd/cluster-agent binaries
  - `omnibus/pkg`: previous omnibus results (not removed by default, and legit to keep around in some cases)
  - `Dockerfiles/`: may contain agent/dogstatsd/cluster-agent artifacts used when working on the dockerfiles

For the last one, one of the limitation of dockerfiles is that they cannot access any file out of the build folder. This means the debian package cannot be retrieved from `../omnibus/pkg/` but must be copied in this folder beforehand

Reply:
I see, thanks for the explanation!

If possible it'd be nice to avoid changing this software definition in such a specific way:
* I'm surprised `bin/` contains anything since I'd expect the binaries to be present in the new folder you mention (i.e. the build folder), instead of the source folder.
* For `omnibus/pkg`, could we play with the omnibus `base_dir` instead so that omnibus writes to a different directory than the agent source's directory?
* For `Dockerfiles/`, I don't see any workaround...

In any case, I don't think it's worth spending too much time on this, I'm fine with keeping this change in the software def as long as the inline comment is a bit more explicit about what the `exclude` option is for, and in which cases it's useful 👍